src/array/shuffle_ext.rs

```diff
@@ -141,11 +133,10 @@ macro_rules! impl_array_impl_shuffle_ext {
                 match self {
                     $(
                         Self::$Abc(builder) => {
-                            let typed_arrays = arrays
-                                .iter()
-                                .map(|x| x.try_into().unwrap())
-                                .collect::<SmallVec<[_; 8]>>();
-                            builder.pick_from_multiple(&typed_arrays, logical_rows);
+                            for (idx, row) in logical_rows {
+                                let arr: & $AbcArray = (&arrays[*idx]).try_into().unwrap();
```

Comment:
This is inefficient because we need to branching inside a loop to do the cast every time. Can we move arr outside?